apps/api/src/core/services/sentry-event/sentry-event.service.ts

```diff
@@ -0,0 +1,92 @@
+import * as Sentry from "@sentry/node";
+import type { Event } from "@sentry/types";
+import type { StackFrame } from "@sentry/types/types/stackframe";
+import { singleton } from "tsyringe";
+
+type ErrorType = Error & {
+  status?: number;
+  statusCode?: number;
+  toJSON?: () => Record<string, any>;
+};
+type ObjectErrorType = Record<string, any>;
+export type ErrorCatchType = ErrorType | ObjectErrorType | string | unknown;
+
+@singleton()
+export class SentryEventService {
```

Comment:
quite often it's not an Error instance that is thrown but a string, or an object. This class makes sure a reasonable event is reported to sentry in any case.